ext/reflection/php_reflection.c

```diff
@@ -2125,7 +2125,137 @@ ZEND_METHOD(ReflectionFunctionAbstract, getNumberOfRequiredParameters)
 }
 /* }}} */
 
-/* {{{ Returns an array of parameter objects for this function */
+/* {{{ Returns whether a parameter exists or not */
+ZEND_METHOD(ReflectionFunctionAbstract, hasParameter)
+{
+    reflection_object *intern;
+    zend_function *fptr;
+    struct _zend_arg_info *arg_info;
+    zval *parameter;
+    uint32_t i, num_args, position;
+
+    if (zend_parse_parameters(ZEND_NUM_ARGS(), "z", &parameter) == FAILURE) {
```

Comment:
The "z" type should be avoided whenever possible. We have a fast ZPP macro Z_PARAM_STR_OR_LONG` for the purpose. not using a generic "z" parameter type is that you can have standard param parsing semantics without any custom code